src/Razor/src/rzls/Program.cs

```diff
@@ -76,16 +76,24 @@ public static async Task Main(string[] args)
         ITelemetryReporter? devKitTelemetryReporter = null;
         if (!telemetryExtensionPath.IsNullOrEmpty())
         {
-            using var exportProvider = await ExportProviderBuilder
-                .CreateExportProviderAsync(telemetryExtensionPath)
-                .ConfigureAwait(true);
+            try
+            {
+                using var exportProvider = await ExportProviderBuilder
+                    .CreateExportProviderAsync(telemetryExtensionPath)
+                    .ConfigureAwait(true);
 
-            // Initialize the telemetry reporter if available
-            devKitTelemetryReporter = exportProvider.GetExports<ITelemetryReporter>().SingleOrDefault()?.Value;
+                // Initialize the telemetry reporter if available
+                devKitTelemetryReporter = exportProvider.GetExports<ITelemetryReporter>().SingleOrDefault()?.Value;
 
-            if (devKitTelemetryReporter is ITelemetryReporterInitializer initializer)
+                if (devKitTelemetryReporter is ITelemetryReporterInitializer initializer)
+                {
+                    initializer.InitializeSession(telemetryLevel, sessionId, isDefaultSession: true);
+                }
+            }
+            catch (Exception ex)
```

Comment:
Is the telemetry DLL the _only_ DLL we get, or will ever want to get, out of MEF? Because if not, this feels scarily generic to me, and could hide any deployment issue. Questions:

* Can we do a File.Exists check on the telemetry DLL, and not add it to the composition if it doesn't exist?
* and/or can we check for a better exception and/or the telemetry DLL filename in the exception message, and rethrow the exception if its not there?

UPDATE: I've read this more closely and it looks like we only use MEF to load the telemetry DLL, and immediately dispose the export provider, which is super weird, but okay. I think its probably worth extracting this bit of code to a method called `InitializeTelemetry` or something, just so we don't accidentally make it a generic "do MEF" in future. If the logic stays targetted at telemetry only, then the pokemon exception handling is probably fine.

Reply:
That's fair. Right now we only do MEF to initialize stuff so... it is what it is. Can definitely help improve this

Reply:
Not for now, but I would think we can do something to make this, and our deployment, easier. Given the code is open source, seems like deploying the DLL always should be no problem, and if we can fix our package to not be self-contained by pick up dependencies from elsewhere, there we could pick up MS.VS.Telemetry from DevKit if needed etc. Anyway, a conversation for another day :)